src/plugins/Wallet/real.ts

```diff
@@ -47,8 +48,37 @@ async function sendTx<R, T extends TransactionObject<R>>(txObject: T, tx: Tx = {
                     gas,
                     gasPrice,
                 })
+                /**
+                 * @returns {String}
+                 *
+                 * Is fired right after the transaction is
+                 * sent and a transaction hash is available.
+                 */
                 .on('transactionHash', (hash: string) => listeners?.onTransactionHash?.(hash))
+                /**
+                 * @returns {TxReceipt}
+                 *
+                 * Is fired when the transaction receipt is available.
+                 */
                 .on('receipt', (receipt: TxReceipt) => listeners?.onReceipt?.(receipt))
+                /**
+                 * @returns {Number, TxReceipt}
+                 *
+                 * Is fired for every confirmation up to the 12th confirmation.
+                 * Receives the confirmation number as the first and the receipt
+                 * as the second argument.
+                 *
+                 * Fired from confirmation 0 on, which is the block where its minded.
+                 */
+                .on('confirmation', (no: number, receipt: TxReceipt) => listeners?.onConfirmation?.(no, receipt))
```

Comment:
This event will emit multi-times, which may cause `onConfirmation` to be invoked many times and encounter the error of resolving a resolved promise.